### PR TITLE
Update azure-client to strict eslint config

### PR DESCRIPTION
**Pull request by tylerbutler**

### CODE REVIEW

File: packages/framework/azure-client/src/test/AzureClient.spec.ts [hunk 1]
```diff
@@ -2,10 +2,12 @@
  * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
  * Licensed under the MIT License.
  */
+// eslint-disable-next-line unicorn/prefer-node-protocol
 import { strict as assert } from "assert";
 import { AttachState } from "@fluidframework/container-definitions";
 import { ContainerSchema } from "@fluidframework/fluid-static";
 import { ISharedMap, IValueChanged, SharedMap } from "@fluidframework/map";
+import type { IFluidHandle } from "@fluidframework/core-interfaces";
```
Comment (skylerjokiel) [inline packages/framework/azure-client/src/test/AzureClient.spec.ts:10]: This is odd. What is forcing this?

File: packages/framework/azure-client/src/AzureClient.ts [hunk 2]
```diff
@@ -139,6 +139,7 @@ export class AzureClient {
             containerSchema,
         );
         const module = { fluidExport: runtimeFactory };
+        // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
         const codeLoader = { load: async () => module };
```
Comment (skylerjokiel) [inline packages/framework/azure-client/src/AzureClient.ts:143]: You can just fix this if you pull in the `ICodeLoader` type from `@fluidframework/container-definitions` 

```suggestion
        const module: ICodeLoader = { fluidExport: runtimeFactory };
        // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
        const codeLoader = { load: async (): ICodeLoader => module };
```

File: packages/framework/azure-client/src/AzureUrlResolver.ts [hunk 3]
```diff
@@ -46,7 +46,8 @@ export class AzureUrlResolver implements IUrlResolver {
             throw new Error("Azure URL did not contain containerId");
         }
         const documentUrl = `${ordererUrl}/${tenantId}/${containerId}`;
-        return Promise.resolve({
+        // unicorn/no-useless-promise-resolve-reject
```
Comment (skylerjokiel) [inline packages/framework/azure-client/src/AzureUrlResolver.ts:49]: Looks like you fixed this. Do you still need the comment?

```suggestion
```

**Review by skylerjokiel — APPROVED** [review]
Added a few nits.

File: packages/framework/azure-client/src/test/AzureClient.spec.ts (hunk 1 above)
Comment (tylerbutler) [reply packages/framework/azure-client/src/test/AzureClient.spec.ts:10]: See the changes on lines 256-257 of this file:

```ts
        const hdl = await map1.get("new-pair-id") as IFluidHandle;
        const obj = hdl.get();
```

Is there a better way to address the "unsafe any" issues in this case?

**Comment by tylerbutler** [comment]
Closing without merging since this review was just to collect feedback.

**Pull request closed without merging**